x/config/configurator_test.go

```diff
@@ -834,6 +850,237 @@ func TestConfigurator(t *testing.T) {
 				return
 			},
 		},
+		{
```

Comment:
Also having a couple of these as integration tests that check the final transport will be useful, but all the interpolation test cases as integration tests might be a little heavy. consider just having unit tests in a `mapdecode_test.go` that tests all the corner cases of interpolation (like what is interpolated, what isn't). they can just check the final struct rather than building a config out of the structs.